### PR TITLE
New Resource azurerm_app_service_hybrid_connection

### PR DESCRIPTION
**Pull request by jackofallops**
Adds support for App Service Hybrid Connections
supersedes #5330

### CODE REVIEW

File: azurerm/internal/services/web/app_service_hybrid_connection_resource.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+			SendKeyValue: utils.String(""), // The service creates this no matter what is sent, but the API requires the field to be set
+		},
+	}
+
+	hybridConnection, err := client.CreateOrUpdateHybridConnection(ctx, resourceGroup, name, namespaceName, relayName, connectionEnvelope)
+	if err != nil {
+		return fmt.Errorf("Error creating App Service Hybrid Connection %q (resource group %q): %s", name, resourceGroup, err)
+	}
+
+	d.SetId(*hybridConnection.ID)
```
Comment (mbfrahry) [inline azurerm/internal/services/web/app_service_hybrid_connection_resource.go:152]: We should confirm this is nil and not empty before setting it

File: azurerm/internal/services/web/tests/app_service_hybrid_connection_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+func TestAccAzureRMAppServiceHybridConnection_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_app_service_hybrid_connection", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMAppServiceHybridConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAppService_basic(data),
```
Comment (mbfrahry) [inline azurerm/internal/services/web/tests/app_service_hybrid_connection_resource_test.go:67]: We'll want to update this
```suggestion
				Config: testAccAzureRMAppServiceHybridConnection_update(data),
```

**Comment by aristosvo** [comment]
General question: is this also working for Function Apps, similar to the Regional VNet Connection?

File: azurerm/internal/services/web/tests/app_service_hybrid_connection_resource_test.go (hunk 2 above)
Comment (jackofallops) [reply azurerm/internal/services/web/tests/app_service_hybrid_connection_resource_test.go:67]: Good spot - ty!

**Comment by jackofallops** [comment]
> General question: is this also working for Function Apps, similar to the Regional VNet Connection?

Hi @aristosvo - afaik the relay doesn't discriminate between app types so should work with Function Apps also, as documented [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-networking-options#:~:text=Hybrid%20Connections%20is%20a%20feature,all%20but%20the%20Consumption%20plan.)

**Comment by schmid37** [comment]
I think my issue #7184 refer to this. I think we can close my issue then.

**Pull request merged**